### PR TITLE
Added the Persistent Volume Snapshots topic

### PR DESCRIPTION
**Pull request by ghost**
Trello card: https://trello.com/c/x3mXFMeX/717-document-volume-snapshot

### CODE REVIEW

**Comment by tmorriso-rh** [comment]
[rev_history]
|xref:../install_config/persistent_storage.adoc[Configuring Persistent Storage]
|Added the xref:.../install_config/persistent_storage.adoc#persistent-volume-snapshots[Persistent Volume Snapshots] section.
%

**Comment by adellape** [comment]
@tmorriso-rh The filename in the `topic_map` correctly uses underscores for the new topic (`persistent_volume_snapshots`), but the actual filename `persistent-volume-snapshots.adoc` is using hyphens. The topic won't actually generate unless the filename matches with underscores.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+  name: snapshot-controller-role
+subjects:
+- kind: ServiceAccount
+  name: snapshot-controller-runner
+  namespace: default
+----
+
+If the external controller and provisioner are deployed in Amazon Web Services
+(AWS), they must be able to authenticate using the access key. To provide the
+credential to the pod, the administrator creates a new secret:
```
Comment (adellape) [inline install_config/persistent_storage/persistent-volume-snapshots.adoc]: These paragraphs and codeblocks that follow are separate from the procedure above it, right?

File: install_config/persistent_storage/persistent-volume-snapshots.adoc [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+snapshots. The provisioner creates new `PersistentVolumes` from the volume
+snapshots. See xref:persistent-volume-snapshots#create-snapshot[Create Snapshot]
+and xref:persistent-volume-snapshots#restore-snapshot[Restore Snapshot] for
+more information.
+
+[[start-controller-and-provisioner]]
+=== Starting the External Controller and Provisioner
+
+The external controller and provisioner services are distributed as container
+images and can be run in the [product-title] cluster as usual. To allow the
```
Comment (adellape) [inline install_config/persistent_storage/persistent-volume-snapshots.adoc]: `[product-title]` should be `{product-title}`.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+
+[NOTE]
+====
+In case of error, the `VolumeSnapshot` status is appended with an `Error` condition.
+====
+
+To display the `VolumeSnapshot` status:
+
+----
+sh
```
Comment (adellape) [inline install_config/persistent_storage/persistent-volume-snapshots.adoc]: Is this `sh` supposed to be the codeblock `source` highlighting type? Or an actual command.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+                  name: awskeys
+                  key: access-key-id
+            - name: AWS_SECRET_ACCESS_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: awskeys
+                  key: secret-access-key
+----
+
+For GCE, there is no need to use secrets to access the AWS cloud provider API.
```
Comment (liangxia) [inline install_config/persistent_storage/persistent-volume-snapshots.adoc]: s/AWS/GCE/

File: install_config/persistent_storage/persistent-volume-snapshots.adoc [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+controller is responsible for creating, deleting, and reporting events on volume
+snapshots. The provisioner creates new `PersistentVolumes` from the volume
+snapshots. See xref:persistent-volume-snapshots#create-snapshot[Create Snapshot]
+and xref:persistent-volume-snapshots#restore-snapshot[Restore Snapshot] for
+more information.
+
+[[start-controller-and-provisioner]]
+=== Starting the External Controller and Provisioner
+
+The external controller and provisioner services are distributed as container
```
Comment (liangxia) [inline install_config/persistent_storage/persistent-volume-snapshots.adoc]: There are also RPM versions for the controller and provisioner.

**Comment by liangxia** [comment]
LGTM. Verified all the yaml content on OCP v3.9.0-0.41.0, and it is working fine.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+
+For example, assume the user 'alice' needs to work with snapshots in the cluster. The cluster administrator completes the following steps: 
+ 
+. Define a new `ClusterRole`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ClusterRole
+metatadata:
```
Comment (liangxia) [inline install_config/persistent_storage/persistent-volume-snapshots.adoc]: Typo of "metadata"

**Comment by liangxia** [comment]
LGTM. Verified all the yaml content on OCP v3.9.0-0.41.0, and it is working fine.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc (hunk 1 above)
Comment (tmorriso-rh) [reply install_config/persistent_storage/persistent-volume-snapshots.adoc]: Yes.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc (hunk 2 above)
Comment (tmorriso-rh) [reply install_config/persistent_storage/persistent-volume-snapshots.adoc]: Fixed.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc (hunk 3 above)
Comment (tmorriso-rh) [reply install_config/persistent_storage/persistent-volume-snapshots.adoc]: Deleted.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc (hunk 4 above)
Comment (tmorriso-rh) [reply install_config/persistent_storage/persistent-volume-snapshots.adoc]: Fixed.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc (hunk 5 above)
Comment (tmorriso-rh) [reply install_config/persistent_storage/persistent-volume-snapshots.adoc]: Added.

File: install_config/persistent_storage/persistent-volume-snapshots.adoc (hunk 6 above)
Comment (tmorriso-rh) [reply install_config/persistent_storage/persistent-volume-snapshots.adoc]: Fixed.

**Comment by tmorriso-rh** [comment]
@adellape Hi Alex. I made your recommended changes. Can you do another review? Thanks!

**Comment by tmorriso-rh** [comment]
@vikram-redhat This pull request has been reviewed (QE/SME/peer) and merged. This needs to be cherry picked to Enterprise 3.9.

**Comment by vikram-redhat** [comment]
@tmorriso-rh - merging this.

**Pull request merged**